Doma APT module-info.java is missing `provides javax.annotation.processing.Processor`

**Describe the bug**

1. Make a Maven project.
1. Make the project modular with `module-info.java`

```
mvn clean install

[ERROR] Bad service configuration file, or exception thrown while constructing Processor object: javax.annotation.processing.Processor: org.seasar.doma.internal.apt.processor.DomainProcessor Unable to get public no-arg constructor
```


**Expected behavior**

For the annotation processors to be found

**Actual behavior**

```
mvn clean install

[ERROR] Bad service configuration file, or exception thrown while constructing Processor object: javax.annotation.processing.Processor: org.seasar.doma.internal.apt.processor.DomainProcessor Unable to get public no-arg constructor
```


**Environment (please complete the following information):**
- Doma version or git rev: 2.54
- IDE version (ie. Eclipse or IntelliJ IDEA): N/A
- Output of `java -version`: 20
- Build tool (ie. output of `mvnw --version` or `gradlew --version`): Apache Maven 3.9.4
- Database version: N/A
- JDBC Driver version: N/A

**Additional context**

```java
module org.seasar.doma.processor {
  requires java.compiler;
  requires org.seasar.doma.core;
}
```

Should be:

```java
module org.seasar.doma.processor {
  requires java.compiler;
  requires org.seasar.doma.core;

  provides javax.annotation.processing.Processor with org.seasar.doma.internal.apt.processor.DomainProcessor,
org.seasar.doma.internal.apt.processor.DataTypeProcessor,
org.seasar.doma.internal.apt.processor.ExternalDomainProcessor,
org.seasar.doma.internal.apt.processor.DomainConvertersProcessor,
org.seasar.doma.internal.apt.processor.EmbeddableProcessor,
org.seasar.doma.internal.apt.processor.EntityProcessor,
org.seasar.doma.internal.apt.processor.DaoProcessor,
org.seasar.doma.internal.apt.processor.ScopeProcessor,
org.seasar.doma.internal.apt.processor.SingletonConfigProcessor;
```

--- a/doma-processor/src/module/org.seasar.doma.processor/module-info.java
+++ b/doma-processor/src/module/org.seasar.doma.processor/module-info.java
@@ -1,4 +1,14 @@
 module org.seasar.doma.processor {
   requires java.compiler;
   requires org.seasar.doma.core;
+  provides javax.annotation.processing.Processor with 
+    org.seasar.doma.internal.apt.processor.DomainProcessor,
+    org.seasar.doma.internal.apt.processor.DataTypeProcessor,
+    org.seasar.doma.internal.apt.processor.ExternalDomainProcessor,
+    org.seasar.doma.internal.apt.processor.DomainConvertersProcessor,
+    org.seasar.doma.internal.apt.processor.EmbeddableProcessor,
+    org.seasar.doma.internal.apt.processor.EntityProcessor,
+    org.seasar.doma.internal.apt.processor.DaoProcessor,
+    org.seasar.doma.internal.apt.processor.ScopeProcessor,
+    org.seasar.doma.internal.apt.processor.SingletonConfigProcessor;
 }

